Tensile/TensileCreateLib/KernelFileContext.py

```diff
@@ -0,0 +1,156 @@
+
+    Notes:
+        - Relies on `openFilesBasedOnFirstKernel` and `openFilesWithFixedNames` for file handling.
+        - The caller is responsible for closing the returned file objects.
+    """
+    kernelSourceFile, kernelHeaderFile = None, None
+    if numMergedFiles > 1 and kernelFiles:
+        kernelSourceFile, kernelHeaderFile = _openFilesBasedOnFirstKernel(kernelFiles)
+    elif mergeFiles or lazyLoading:
+        kernelSourceFile, kernelHeaderFile = _openFilesWithFixedNames(outputPath)
+
+    return kernelSourceFile, kernelHeaderFile
+
+
+def _closeKernelFiles(
+    kernelSourceFile: Optional[TextIOWrapper], kernelHeaderFile: Optional[TextIOWrapper]
+):
+    """Closes the kernel source and header file objects if they are open.
+
+    This function checks if the provided file objects for the kernel source and header files are
+    not None (they are open) and closes them. It's a cleanup function to ensure that file resources
+    are properly released after their use.
+
+    Args:
+        kernelSourceFile: The file object for the kernel's source code, or None if it's not open.
+        kernelHeaderFile: The file object for the kernel's header code, or None if it's not open.
+    """
+    if kernelSourceFile:
+        kernelSourceFile.close()
+    if kernelHeaderFile:
+        kernelHeaderFile.close()
+
+
+def _openFilesWithFixedNames(outputPath: Path) -> Tuple[TextIOWrapper, TextIOWrapper]:
+    """Opens files for kernel source code (Kernels.cpp) and header code (Kernels.h).
+
+    Args:
+        outputPath: The directory path where the files should be opened.
+
+    Returns:
+        A tuple containing two `TextIOWrapper` objects for the source and header files respectively.
+
+    Notes:
+        - The source file is named "Kernels.cpp" and the header file is named "Kernels.h".
+        - Both files are opened in append mode with UTF-8 encoding.
+    """
+    srcFilename = Path(outputPath).resolve() / "Kernels.cpp"
+    hdrFilename = Path(outputPath).resolve() / "Kernels.h"
+    return open(srcFilename, "a", encoding="utf-8"), open(hdrFilename, "a", encoding="utf-8")
+
+
+def _openFilesBasedOnFirstKernel(kernelFiles: List[str]) -> Tuple[TextIOWrapper, TextIOWrapper]:
+    """Opens two files for appending based on the name of the first kernel file in the list.
+
+    Args:
+        kernelFiles: A list of kernel file names, where the first element is used to determine the base file name.
+
+    Returns:
+        A tuple containing two `TextIOWrapper` objects for appending to the source and header files respectively.
+
+    Notes:
+        - The base file name is derived by removing the ".cpp" extension from the first kernel file name.
+        - The source and header files are named using the base name with ".cpp" and ".h" extensions respectively.
+        - Files are opened in append mode with UTF-8 encoding.
+    """
+    filename = kernelFiles[0].replace(".cpp", "")
+    return open(filename + ".cpp", "a", encoding="utf-8"), open(
```

Comment:
The wrapping of open is a little awkward.

Reply:
This is black's default formatting. I don't love it either, but still support using automated formatting even if sometimes it feels a little awkward on edge cases.